### PR TITLE
chore(a2a): upgrade Java SDK to 1.0.0.Final

### LINKED ISSUE #4762: [Dependency Upgrade] Upgrade a2a-java-sdk from 0.2.5.Beta2 to 1.0.0.Final
### Feature Request

需求背景
当前项目依赖 a2a-java-sdk 版本为 0.2.5.Beta2，该版本为测试 Beta 版，存在如下问题：
Beta 版本稳定性不足，线上偶现接口序列化、连接池异常问题；
缺少 1.0.0.Final 版本新增的鉴权、批量请求、异常重试能力；
官方已停止维护 0.2.x Beta 分支，1.0.0.Final 为正式稳定发行版，长期支持。
升级目标
将项目所有模块的 a2a-java-sdk 统一升级至稳定正式版：1.0.0.Final
升级影响评估
API 兼容校验
查阅官方变更日志：1.0.0.Final 存在少量不兼容调整：
部分实体类包路径重构；
同步 / 异步客户端构造器参数变更；
错误码枚举字段重命名。
改造工作量
全局替换依赖版本号；
调整客户端初始化代码；
修正实体导入、异常捕获逻辑；
单元测试、集成测试全量回归。
风险
短期存在代码改动成本，升级完成后消除 Beta 版本线上隐患，长期降低维护成本

### Is your feature request related to a problem? Please describe

_No response_

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by March-77**
## Summary

- upgrade the A2A Java SDK from `0.2.5.Beta2` to `1.0.0.Final` and migrate to the `org.a2aproject.sdk` coordinates
- adapt agent cards, JSON-RPC request/response handling, task emission, transport discovery, and well-known routes to the v1 SDK API
- preserve Nacos interoperability for legacy agent-card interfaces and security schemes, and remove the previous dummy extended-card credential
- preserve JSON-RPC request ids in post-parse handler errors and send the selected interface's `A2A-Version` header on both non-streaming and streaming requests
- add regression coverage for v1 request payloads, wrapped responses, request-id preservation, A2A version headers, Spring server startup, and Nacos conversion

## Verification

- `./mvnw -pl :spring-ai-alibaba-starter-a2a-nacos -am -Dtest=A2aRemoteAgentTests,A2aNodeActionWithConfigTests,Issue3608ReproductionTest,A2aServerHandlerAutoConfigurationTest,A2aServerMultiAgentAutoConfigurationTest,NacosAgentCardWrapperTest,NacosA2aPropertiesTest,AgentCardConverterUtilTest,JsonRpcA2aRequestHandlerTest -Dsurefire.failIfNoSpecifiedTests=false -Dcheckstyle.skip=true test` (48 tests)
- `./mvnw -pl :spring-ai-alibaba-starter-a2a-nacos -am -DskipTests=true test` (compilation, Spotless, and Checkstyle; 0 violations)

Closes #4762

### CODE REVIEW

**Comment by yuluo-yx** [comment]
可以看下 ai 的评审内容，修改下

**Comment by March-77** [comment]
已根据 AI 评审完成修改（commit `613526673`）：

- 服务端在请求解析成功后的 `A2AError` 和内部错误响应中保留原 JSON-RPC request ID，并覆盖非流式、流式两条处理路径。
- 客户端从选中的 `AgentInterface` 读取协议版本，在非流式和流式 POST 中统一设置 `A2A-Version`。
- 相关 A2A 回归测试共 48 项通过；编译、Spotless 和 Checkstyle 也已通过（0 violations）。

@codex review

File: spring-boot-starters/spring-ai-alibaba-starter-a2a-nacos/src/main/java/com/alibaba/cloud/ai/a2a/core/server/GraphAgentExecutor.java [hunk 1]
```diff
@@ -62,75 +60,71 @@ public class GraphAgentExecutor implements AgentExecutor {
...
 	}
 
 	@Override
-	public void cancel(RequestContext context, EventQueue eventQueue) throws JSONRPCError {
+	public void cancel(RequestContext context, AgentEmitter emitter) throws A2AError {
+		String streamKey = streamKey(context, emitter);
+		StreamExecution execution = this.activeStreams.get(streamKey);
+		if (execution == null) {
+			emitter.cancel();
+			return;
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-boot-starters/spring-ai-alibaba-starter-a2a-nacos/src/main/java/com/alibaba/cloud/ai/a2a/core/server/GraphAgentExecutor.java:109]: Addressed in `be7bad98d`. Non-stream executions are now tracked with a terminal-state arbiter. Cancellation interrupts and briefly waits for the worker, claims the cancelled terminal state atomically, and prevents the invocation from publishing late artifacts, completion, or failure events. `cancelPreventsLateNonStreamArtifactsAndCompletion` covers an invocation that ignores interruption until released. The full A2A module suite passes (52 tests), with Checkstyle and Spotless clean.

File: spring-boot-starters/spring-ai-alibaba-starter-a2a-nacos/src/main/java/com/alibaba/cloud/ai/a2a/autoconfigure/client/A2aClientAgentCardProviderAutoConfiguration.java [hunk 2]
```diff
@@ -50,25 +57,51 @@ public AgentCardProvider remoteAgentCardProvider(A2aClientAgentCardProperties a2
...
+			if (protocolVersion == null || protocolVersion.isBlank()) {
+				protocolVersion = A2aConstants.DEFAULT_A2A_PROTOCOL_VERSION;
+			}
+			interfaces.add(new AgentInterface(transport, properties.getUrl(), null, protocolVersion));
+		}
+		if (properties.getAdditionalInterfaces() != null) {
+			properties.getAdditionalInterfaces()
+				.stream()
+				.filter(agentInterface -> !interfaces.contains(agentInterface))
+				.forEach(interfaces::add);
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-boot-starters/spring-ai-alibaba-starter-a2a-nacos/src/main/java/com/alibaba/cloud/ai/a2a/autoconfigure/client/A2aClientAgentCardProviderAutoConfiguration.java:103]: Good catch. Addressed in `d2f8501b3`: additional interfaces now normalize a null or blank `protocolVersion` to `A2aConstants.DEFAULT_A2A_PROTOCOL_VERSION` while preserving protocol binding, URL, and tenant. Added regression coverage for an additional-only JSON-RPC interface with no configured version. The focused test passes 2/2 and the full A2A module suite passes 52/52; Checkstyle and Spotless are clean.

**Comment by March-77** [comment]
已检查并处理当前 head 上仍有效的评审意见，更新在 `12a749b19`：

- 流式 `hookOnNext` 现在通过 `StreamExecution` 的同一终止锁发布 artifact；取消会等待正在执行的输出结束，之后才发送 terminal cancel，避免取消状态之后继续产生 artifact。
- 新增并发回归测试，刻意阻塞 `addArtifact` 后并发取消，验证信号顺序严格为 artifact → cancel。
- 文档示例中的旧 `/.well-known/agent.json` 已更新为 `/.well-known/agent-card.json`，同时把旧 `/a2a/message` 更新为当前 `/a2a`。
- server additional interfaces 的缺省协议版本问题在当前 head 已经处理：单 Agent 与多 Agent 路径都调用 `A2aAgentInterfaceNormalizer.withDefaultProtocolVersion`，现有两组回归测试覆盖缺省值和显式版本保留。

验证：`GraphAgentExecutorTest`、`A2aServerHandlerAutoConfigurationTest`、`A2aServerMultiAgentAutoConfigurationTest` 共 20 个测试通过；A2A 模块 Spotless 通过，`git diff --check` 通过。

完整 reactor 构建仍会在与本 PR 无关的 `graph-core/OracleSaverTest.java:487` 既有语法错误处提前失败，因此本次使用 JDK 25（`--release 17`）直接验证受影响的 A2A 模块。

File: spring-boot-starters/spring-ai-alibaba-starter-a2a-nacos/src/main/java/com/alibaba/cloud/ai/a2a/autoconfigure/server/A2aServerAgentCardAutoConfiguration.java [hunk 3]
```diff
@@ -116,12 +117,24 @@ private List<AgentSkill> getAgentSkills(Agent rootAgent,
...
-		if (null != a2AServerAgentCardProperties.getAdditionalInterfaces()) {
-			return a2AServerAgentCardProperties.getAdditionalInterfaces();
+		List<AgentInterface> interfaces = new ArrayList<>();
+		interfaces.add(new AgentInterface(a2aServerProperties.getType(),
+				getUrl(a2aServerProperties, a2AServerAgentCardProperties)));
+		if (a2AServerAgentCardProperties.getAdditionalInterfaces() != null) {
+			a2AServerAgentCardProperties.getAdditionalInterfaces()
+				.stream()
+				.filter(agentInterface -> !interfaces.contains(agentInterface))
+				.forEach(interfaces::add);
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-boot-starters/spring-ai-alibaba-starter-a2a-nacos/src/main/java/com/alibaba/cloud/ai/a2a/autoconfigure/server/A2aServerAgentCardAutoConfiguration.java:130]: Verified on the current head. Both single-agent and multi-agent server paths normalize null or blank additional-interface versions through `A2aAgentInterfaceNormalizer` while preserving explicit versions. The client, single-server, and multi-server focused suites pass 8/8.

File: spring-boot-starters/spring-ai-alibaba-starter-a2a-nacos/src/main/java/com/alibaba/cloud/ai/a2a/core/route/JsonRpcA2aRouterProvider.java [hunk 4]
```diff
@@ -43,7 +41,7 @@ public class JsonRpcA2aRouterProvider implements A2aRouterProvider<JsonRpcA2aReq
 
 	private static final Logger log = LoggerFactory.getLogger(JsonRpcA2aRouterProvider.class);
 
-	public static final String DEFAULT_WELL_KNOWN_URL = "/.well-known/agent.json";
+	public static final String DEFAULT_WELL_KNOWN_URL = "/.well-known/agent-card.json";
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-boot-starters/spring-ai-alibaba-starter-a2a-nacos/src/main/java/com/alibaba/cloud/ai/a2a/core/route/JsonRpcA2aRouterProvider.java:44]: Completed in the current head, with the remaining stale invocation route fixed in `e94fc8fe5`. The advanced example now uses `/.well-known/agent-card.json` for the card and `/a2a` for JSON-RPC calls in both README and console guidance. The standalone documentation module compiles successfully.

**Comment by March-77** [comment]
Fixed the post-merge codespell failure in `658595558` by using spellcheck-safe stream snapshot fixtures without changing the delta/replacement semantics. `A2aNodeActionWithConfigTests` passes 37/37 with Checkstyle and Spotless clean.

**Comment by March-77** [comment]
Fixed the full-test cancellation race in `a3b3c6e77`. A producer that passed the pre-enqueue cancellation check can no longer repopulate the queue after cancellation clears it; the post-enqueue check now restores the single terminal sentinel before exiting. The failing cancellation regression passes 5 consecutive runs, and the full focused class passes 37/37.

**Review by oss-sentinel-ai — COMMENTED** [review]
### Summary
Reviewed the changes in this PR (6541 lines changed). The changes appear to be well-structured.

### Observations
- Changes span 6541 lines
- Please ensure all edge cases are covered
- Verify test coverage for the modified code paths

---
*Automated review by github-manager-bot*

**Comment by March-77** [comment]
@yuluo-yx 已重新核对当前 head、最新 main 变化、mergeability、CI、reviews 和 review threads；现有评审意见均已验证并清理，未发现需要追加代码修改的问题。烦请在方便时进行维护者审查，谢谢。

**Comment by KoJun233** [comment]
这个补丁近期有计划合并么？ 我也遇到了这样的问题，似乎0.25和1.0的协议不支持
